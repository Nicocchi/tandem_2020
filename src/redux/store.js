import { createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import logger from "redux-logger";
import combineReducer from "./reducers";

let store = null;

// Redux dev tools
const composeEnhancers =
    typeof window === "object" && window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__
        ? window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__({
              // Specify extensionâ€™s options like name, actionsBlacklist, actionsCreators, serialize...
          })
        : compose;

const devEnhancer = composeEnhancers(applyMiddleware(thunk, logger));
const enhancer = composeEnhancers(applyMiddleware());

// Create redux store
if (process.env.REACT_APP_NODE_ENV !== "production") {
    store = createStore(combineReducer, devEnhancer);
} else {
    store = createStore(combineReducer, enhancer);
}

export default store;