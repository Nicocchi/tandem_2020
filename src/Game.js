import React, { useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from "react-redux";
import QuestionBox from './components/QuestionBox/QuestionBox';
import shortid from "shortid";
import { setFinished } from "./redux/actions";
import ScoreBox from './components/ScoreBox/ScoreBox';

function Game(props) {
    const [currentQuestion, setCurrentQuestion] = useState(0);
    const [endAmount, setEndAmount] = useState(0);

    const { questions } = props;

    useEffect(() => {
        if (questions && questions.length > 0) {
            setEndAmount(questions.length);
        }

        return () => {
            
        }
    }, [questions])

    const nextQuestion = () => {
        let endA = endAmount;
        let currentQuestionA = currentQuestion;

        if (currentQuestion === endA - 1) {
            return props.setFinished();
        }

        if (currentQuestion !== endA - 1) {
            setCurrentQuestion(currentQuestionA += 1);
        }
    }

    const playAgain = () => {
        props.history.push("/");
    }

    if (!props.isPlaying) {
        props.history.push("/");
    }

    return (
        <div>
            <h1>Game</h1>
            {
                !props.isFinished ? questions.map((question, index) => (
                    index === currentQuestion ? <QuestionBox key={shortid.generate()} question={question} nextQuestion={nextQuestion} /> : null
                )) : null
            }
            {
                props.isFinished ? <ScoreBox playAgain={playAgain} /> : null
            }
        </div>
    )
}

Game.propTypes = {
    questions: PropTypes.array,
    isFinished: PropTypes.bool,
    isPlaying: PropTypes.bool,
}

const mapStateToProps = (state) => {
    return {
      questions: state.gameReducer.questions,
      isFinished: state.gameReducer.isFinished,
      isPlaying: state.gameReducer.isPlaying,
    }
  }
  
  export default connect(mapStateToProps, { setFinished })(Game);

