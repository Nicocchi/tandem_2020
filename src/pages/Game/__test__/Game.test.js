import React from "react";
import ReactDOM from "react-dom";
import Game from "./../Game";
import store from "../../../redux/store";
import { Provider } from "react-redux";
import { Router, Switch } from "react-router-dom";
import { createMemoryHistory } from "history";

import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

describe("Game Page", () => {
    const history = createMemoryHistory({ initialEntries: ["/"] });

    afterEach(() => {
        cleanup();
    });

    it("renders without crashing without props", () => {
        const div = document.createElement("div");

        ReactDOM.render(
            <Provider store={store}>
                <Router history={history}>
                    <Game history={history} />
                </Router>
            </Provider>,
            div
        );
    });

    it("matches snapshop", () => {
        const tree = renderer
            .create(
                <Provider store={store}>
                    <Router history={history}>
                        <Game history={history} />
                    </Router>
                </Provider>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
