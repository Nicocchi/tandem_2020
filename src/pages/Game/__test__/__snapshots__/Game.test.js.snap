// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Game Page matches snapshop 1`] = `
<div
  className="game-wrapper"
/>
`;
